### PR TITLE
New flag to set site replication to synchronous

### PR DESCRIPTION
**Pull request by feorlen**
Document new flag for `mc admin replicate update` to enable/disable synchronous site replication. Continue to recommend default async replication due to increased write latency with sync.

Adapt/reuse the existing description of async vs sync from bucket replication as an include.

Staged
http://192.241.195.202:9000/staging/DOCS-790/linux/html/operations/install-deploy-manage/multi-site-replication.html#synchronous-vs-asynchronous-replication
http://192.241.195.202:9000/staging/DOCS-790/linux/html/reference/minio-mc-admin/mc-admin-replicate.html#syntax
http://192.241.195.202:9000/staging/DOCS-790/linux/html/administration/bucket-replication.html#synchronous-vs-asynchronous-replication

Fixes https://github.com/minio/docs/issues/790

### CODE REVIEW

File: source/reference/minio-mc-admin/mc-admin-replicate.rst [hunk 1]
```diff
@@ -80,6 +80,8 @@ Syntax
    :fullpath:
 
    Create or expand a site replication configuration.
+   New configurations use asynchronous site replication.
+   To enable synchronous replication for an existing configuration, see :mc-cmd:`~mc admin replicate update --sync`.
```
Comment (feorlen) [inline source/reference/minio-mc-admin/mc-admin-replicate.rst:84]: I'm meh about this wording for these two sentences. Suggestions appreciated.

File: source/reference/minio-mc-admin/mc-admin-replicate.rst [hunk 2]
```diff
@@ -306,7 +315,7 @@ Syntax
                                 minio1          \
                                 --bucket images
 
-         Display the site repliction status for the setting for a user, ``janedoe``, across sites for a site replication configuration that contains the site ``minio1``.
+         Display the site replication status for the setting for a user, ``janedoe``, across sites for a site replication configuration that contains the site ``minio1``.
```
Comment (feorlen) [inline source/reference/minio-mc-admin/mc-admin-replicate.rst:318]: One of several unrelated spelling fixes.

**Review by ravindk89 — APPROVED** [review]
One nit, otherwise LGTM

File: source/reference/minio-mc-admin/mc-admin-replicate.rst (hunk 1 above)
Comment (djwfyi) [reply source/reference/minio-mc-admin/mc-admin-replicate.rst:84]: > New site replication configurations use asynchronous mode by default, as MinIO recommends.
> If your circumstances require synchronous replication, complete the configuration, then use :mc-cmd:`mc admin replicate update --sync`.

File: source/reference/minio-mc-admin/mc-admin-replicate.rst [hunk 3]
```diff
@@ -176,6 +178,13 @@ Syntax
       
       The new endpoint or URL to associate with the peer site.
 
+   .. mc-cmd:: --sync
+      :optional:
+
+      Enable or disable synchronous site replication.
+      Available values are ``enable`` and ``disable``.
+      The default is ``disable`` to use asynchronous site replication.
```
Comment (djwfyi) [inline source/reference/minio-mc-admin/mc-admin-replicate.rst:186]: > When not defined, MInIO uses asynchronous site replication.

**Review by djwfyi — APPROVED** [review]
Some wording suggestions you can take or leave.
Otherwise, LGTM.

File: source/includes/common-replication.rst [hunk 4]
```diff
@@ -107,6 +107,20 @@ See :mc:`mc admin user`, :mc:`mc admin user svcacct`, and :mc:`mc admin policy`
...
+.. start-replication-sync-vs-async
+
+MinIO supports specifying either asynchronous (default) or synchronous replication for a given remote target.
+
+With the default asynchronous replication, MinIO completes the originating ``PUT`` operation *before* placing the object into a :ref:`replication queue <minio-replication-process>`.
+The originating client may therefore see a successful ``PUT`` operation *before* the object is replicated.
+While this may result in stale or missing objects on the remote, it mitigates the risk of slow write operations due to replication load.
+
+With synchronous replication, MinIO attempts to replicate the object *prior* to completing the originating ``PUT`` operation.
+MinIO returns a successful ``PUT`` operation whether or not the replication attempts succeeds.
```
[the remark this answers is not in the document]
Comment (feorlen) [reply source/includes/common-replication.rst:119]: Ha completely missed that in the original. 👍🏻

File: source/reference/minio-mc-admin/mc-admin-replicate.rst (hunk 1 above)
Comment (feorlen) [reply source/reference/minio-mc-admin/mc-admin-replicate.rst:84]: Went with a mashup of the two, as `add` doesn't have `--sync`. Only `update` does.

Comment (ravindk89) [reply source/reference/minio-mc-admin/mc-admin-replicate.rst:84]: Completely slipped past me. That's an interesting design choice.

**Review by poornas — APPROVED** [review]
